src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,429 @@
+---
+title: "Codelab: Basic Flutter layout concepts"
+description: "A codelab that teaches basic Flutter layout concepts through DartPad2 code examples."
+toc: true
+--- 
+{{site.alert.important}}
+  This codelab covers basic Flutter layout concepts using a code editor called DartPad2. An understanding of [Dart](https://www.dart.dev) or some exposure to object-oriented programming is recommended. 
+{{site.alert.end}}
+{{site.alert.important}}
+  Currently, DartPad2 is in the experimental stage and being tested with Google Chrome. A PC (Microsoft, Mac, or Linux) with Google Chrome installed is required for this codelab. If you encounter any difficulties while using DartPad2, please create a [DartPad issue](https://github.com/dart-lang/dart-pad/issues/new) and label the issue "platform-web."
+{{site.alert.end}} 
+{{site.alert.note}}
+  Flutter supports devices that maintain right-to-left language settings. 
+{{site.alert.end}}
+## Introduction
+
+In this codelab, you'll interact with and build Flutter layouts in a code editor called DartPad2. 
+As you progress through this codelab, you learn that widgets build everything in Flutter. 
+A widget is an immutable object that describes a specific part of a UI. You also learn that, 
+while widgets are simple in their functions, widgets can be complex in their structures. 
+Widgets can be built around and inside of other widgets, 
+and widgets can have interconnected and ordered relationships with one another. At the end of this codelab, 
+you'll apply what you learn and build a simple interface using basic Flutter layout concepts. 
+## Row and Column classes
+`Row` and `Column` are classes that contain and lay out widgets. These widgets are called "children," 
+and `Row` and `Column` are referred to as "parents." `Row` lays out its widgets *horizontally*. 
+`Column` lays out its widgets *vertically*. 
+
+#### Code example: Row
+{:.no_toc}
+{{site.alert.secondary}}
+{:.no_toc}
+  Check out Dart.
+  
+  The following code example displays three blue squares horizontally. 
+
+  Click the **Run** button.
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/26c8fb02c58c4120917b073ec6253953
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=26c8fb02c58c4120917b073ec6253953&amp;fw=true&amp;split=60" width="100%" height="400px"></iframe> 
+
+#### Code example: Column
```

Comment:
Changed. I also changed the one from "Row" to "Create a Row." I just wonder if these headings will change once I change the color of the DartPads because these examples are two out of three where the user doesn't do anything but run the DartPads.